### PR TITLE
issue #426 fix for pipeline derby jdbc test error

### PR DESCRIPTION
**Pull request by albertwang-ibm**
The fixes:
(1) To speed up sequence fetch, add a derby config to pre-create 1000 sequence numbers instead of the default 100.
(2) Made changes to make sure committing the transaction after the processing of each example, then old codes commit only once after all the test examples are processed.

### CODE REVIEW

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java [hunk 1]
```diff
@@ -92,7 +115,8 @@ protected FHIRPersistenceContext createHistoryPersistenceContext() {
 
     @Override
     public FHIRPersistence getPersistenceImpl() throws Exception {
-        return new FHIRPersistenceJDBCImpl(this.properties, database);
+        //return new FHIRPersistenceJDBCImpl(this.properties, database);
```
Comment (prb112) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: what's this commented for?

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java [hunk 2]
```diff
@@ -44,10 +48,29 @@ public void bootstrapAndLoad() {
...
+        operations.add(new HistoryOperation(3));
+        operations.add(new DeleteOperation());
+        operations.add(new DeleteOperation());
+        operations.add(new HistoryOperation(4));
+        R4JDBCExamplesProcessor processor = new R4JDBCExamplesProcessor(
+                operations,
+                this.properties,
+                connectionPool,
+                null,
+                null,
```
Comment (prb112) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: is it necessary to have null here? on 71/27?

Comment (prb112) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: maybe cleanup the class?

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java [hunk 3]
```diff
@@ -47,18 +55,28 @@
      */
     public DerbyMaster(String database) {
         this.database = database;
-                
-        try {
+
+        try (PrintWriter out = new PrintWriter(DERBY_PROPERTIES)){
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java]: if I understand this correctly, the following entries are being written to the derby properties file?

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java [hunk 4]
```diff
@@ -47,18 +55,28 @@
...
             Class.forName(DERBY_TRANSLATOR.getDriverClassName());
+            // This speeds up sequence fetching by pre-creating 1000 instead of the default 100.
+            out.println("derby.language.sequence.preallocator=1000");
+            if (debugging) {
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java]: I like this...+1

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java (hunk 1 above)
Comment (albertwang-ibm) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: good catch, I mean to remove it. will remove

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java (hunk 2 above)
Comment (albertwang-ibm) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: this is a shared function used by both db2 and derby, used to pass tenant names and key for db2 test.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java (hunk 3 above)
Comment (albertwang-ibm) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyMaster.java]: yes, this is exactly what this does.

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java (hunk 1 above)
Comment (albertwang-ibm) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/spec/R4JDBCExamplesTest.java]: done

**Comment by prb112** [comment]
Executed re-run checks.

**Pull request merged**